### ISSUE TITLE
Should we be building in Next.js? 

### ISSUE THREAD

**Issue by fliebenberg** [DeXter-on-Radix/website#42]
Next.js is not ideal for an app that wants to run completely on the client. We will also not use its other features like routing since our app structure should be pretty simple. And it is causing issues by always trying to render on the server.

I am just wondering if it wont be better to just switch to normal react or even another language that is more suited for what we want to do.

It is still early enough if we do want to switch. The further we progress the harder it will be to switch later.

---

## Alternatives to try out with radix-dapp-toolkit

- [ ] [create-react-app](https://create-react-app.dev/)
- [ ] [svelte-kit](https://kit.svelte.dev/)
- [ ] [gatsby](https://www.gatsbyjs.com/)
- [ ] (maybe) [astro](https://docs.astro.build/en/concepts/why-astro/) but it's MPA so... maybe not our thing

**Comment by EvgeniiaVak** [comment]
I think we will use routing, if we have more than one page. For example we are thinking on the landing page currently.

**Comment by EvgeniiaVak** [comment]
It should be possible to fix the document-not-defined issue (probably prioritize #4 I'll look into it today) - next.js has SSG (static site generation - exactly what we need). That said if you know about a framework that suits us better, I don't mind switching to it. I don't particularly like react, I'd vote for svelte-kit (but I have very little experience with it).

**Comment by fliebenberg** [comment]
It would be great if you could have a go at it. I published a branch called "fix: build error". I moved all the setting up of createContext into a separate file, which solves the multiple exports problem.

If you look at it, we are not using any functionality of next.js yet. Everything is basically react so far. react does have routing, you just have to do it through installed packages. I am just not sure we are getting any benefits from using next.js and it is causing some frustration - it is so weird that they make it so difficult to just skip rendering on the server. You will find that SSG does not really solve the problem. It still generates the static parts of the page on the server and that then throws the error we are encountering. But I do hope you succeed so we can move on. 

Shout if you have questions about my changes or just need some help. Maybe we can all solve this together.

**Comment by EvgeniiaVak** [comment]
The problem with `document` being not defined stems from the fact that "@radixdlt/radix-dapp-toolkit" lib uses it quite a lot in a not server-compatible way (btw this will become a problem for this lib in the future, SSR and SSG are quite popular things). To fix it and still use next.js we would need to wrap all the imports with something like:
```TS
import dynamic from 'next/dynamic';

const RadixDappToolkit = dynamic(() => import('@radixdlt/radix-dapp-toolkit'), {
  ssr: false, // This will load the module only on client-side
});
```
which is ugly and we'd only learn that we need to do it during build time, not convenient at all.

SSG is good for SEO and performance, if we won't have it, we might have this nasty lag with an empty page page for a sec before the scripts are fetched and completed running. But we can think about it later I guess, and maybe use some pre-renderer that uses a browser instead of node.

So I agree - we will probably be better off ditching next.js and using another lib for routing when we get to multiple pages 🚀 

Off topic:
Also I noticed in the toolkit lib there are cookie set... that's a pity, I was hoping our website wouldn't to touch cookies and thus could be able to avoid dealing with GDPR.

**Comment by EvgeniiaVak** [comment]
Ah, another everyday thing we use from next.js is the dev server, if we remove that we would need to setup something else instead, with hot reload and all that. Also the bundler (webpack) came pre-configured. 
Given that we have problems only during build time, I suggest we finish #6 and only than do refactoring to replace next.js, so there would be less code conflicts. In the meantime we could test how different alternatives behave in tandem with radix-dapp-toolkit

**Comment by EvgeniiaVak** [comment]
Relevant answer in [Radix discord](https://discord.com/channels/417762285172555786/1018955196786823188/1055865893873258646):

> The problem is that you're running the connector button on the server during SSR. The button only works in the browser, as it uses browser specific API's.
>
> Put the configure in an onMount() and it will work. You can also disable SSR for that page by putting export const ssr = false inside your +page.js file. 

they talk about svelte, but we can apply a similar principle, it looks like it's possible to use the radix button with SSG properly, looking into it

**Comment by fliebenberg** [comment]
About the cookie: they use it to automatically link the chosen accounts and persona to an app so that the client does not have to connect the wallet every time they come back to the app. I dont think it is too much of an issue for most people (they like the convenience), but yea, we will have to disclose it on the app.

**Comment by EvgeniiaVak** [comment]
I suggest we keep all the initializations in useEffect and keep SSG (I don't mind migrating from Next.js, but from what I've seen the other frameworks work similar by pre-rendering even in build time on server). I hope SSG will help us to build a sleek app that loads the view asap for the user (the idea is to render all the html elements instantaneously and when all the init scripts are done - fill them with life). It does require a little understanding of what happens when and a few checks for undefined etc.

document not defined fix #44

**Comment by EvgeniiaVak** [comment]
closing this as we made the build working with SSG in #46
